packages/jest/src/utils/config/get-jest-projects.ts

```diff
@@ -0,0 +1,46 @@
+import { readWorkspaceConfig } from '@nrwl/workspace/src/core/file-utils';
+import { dirname, join } from 'path';
+import type { WorkspaceJsonConfiguration } from '@nrwl/devkit';
+
+const JEST_RUNNER_TOKEN = '@nrwl/jest:jest';
+
+function getJestConfigProjectPath(projectJestConfigPath: string): string {
+  return join('<rootDir>', dirname(projectJestConfigPath));
+}
+
+export function getJestProjects() {
+  const ws = readWorkspaceConfig({
+    format: 'nx',
+  }) as WorkspaceJsonConfiguration;
+  const jestConfigurationSet = new Set();
+  for (const projectConfig of Object.values(ws.projects)) {
+    if (!projectConfig.targets) {
+      continue;
+    }
+    for (const targetConfiguration of Object.values(projectConfig.targets)) {
+      if (
+        targetConfiguration.executor === JEST_RUNNER_TOKEN &&
+        targetConfiguration.options?.jestConfig
+      ) {
+        jestConfigurationSet.add(
+          getJestConfigProjectPath(targetConfiguration.options.jestConfig)
+        );
+      }
+      if (targetConfiguration.configurations) {
+        for (const configurationObject of Object.values(
+          targetConfiguration.configurations
+        )) {
+          if (configurationObject.executor === JEST_RUNNER_TOKEN) {
+            jestConfigurationSet.add(
+              getJestConfigProjectPath(
+                configurationObject.options?.jestConfig ||
+                  targetConfiguration.options.jestConfig
```

Comment:
You shouldn't need to fallback to the default options. Those were already added above so this `add` would never actually add anything if the configuration doesn't have a `jestConfig` defined.

Reply:
Makes sense - I was attempting to cover for a scenario where a user wanted to run either karma or jest via a configuration option (obscure enough already). The thought being that the user could set a `jestConfig` option either at the target level (even if the original target wasn't using our jest runner) OR at the configuration option (with the specific configuration option taking priority). I think obscure enough that should be irrelevant.